sktime/datatypes/_vectorize.py

```diff
@@ -593,7 +593,18 @@ def _to_iter(e):
             ret.append((group_name, col_name, est_i_result))
 
         if return_type == "pd.DataFrame":
-            df = pd.DataFrame(ret).pivot(index=0, columns=1, values=2)
+            df_long = pd.DataFrame(ret)
+            cols_right_order = df_long.loc[:, 1].unique()
+
+            df = df_long.pivot(index=0, columns=1, values=2)
+            # DataFrame.pivot sorts the columns (is this a bug? see #4683)
+            # either way, we need to fix this:
+            df = df.loc[:, cols_right_order]
+
+            # remove "0" and "1" from index/columns name
+            df.index.names = [None] * len(df.index.names)
+            df.columns.name = None
+
```

Comment:
What about this? pivot supports a sort arg since 1.3 (https://github.com/pandas-dev/pandas/pull/40954)

```suggestion
            df = pd.DataFrame(ret)
            try:  # pandas >= 1.3
                df = df.pivot(index=0, columns=1, values=2, sort=False)
            except TypeError:
                df = df.pivot(index=0, columns=1, values=2).reindex(df[1], axis=1)
```

Reply:
Ah, sorry. That only applies to pivot_table. So it would be just 

```suggestion
            df = pd.DataFrame(ret)
            df = df.pivot(index=0, columns=1, values=2).reindex(df[1], axis=1)
```

Reply:
ah, thanks - that's much nicer.

Quick questions:

* would that work on `pandas` 1.1 too? we try to be downwards compatible where that is still possible
* there's also the matter of the spurious column/level labels "0" and "1" that appear through using `pivot`, they should optimally be replaced by the labels that were there to start with in the data, or removed
    * for an example of this malfunction (bit ugly but ultimatly inconsequential), search for `forecasters_` in this notebook: https://github.com/sktime/sktime-tutorial-ODSC-Europe-2023/blob/main/notebooks/03_forecasting.ipynb

Reply:
thanks for the swift review btw! 🚀